staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -302,50 +317,44 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// with non-equal specs (especially non-equal schemas).
 		var crds []*apiextensionsv1.CustomResourceDefinition
 		crds, err = r.crdLister.List(labels.Everything())
-		if err == nil {
-			if len(crds) == 0 {
-				err = errors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, "")
+		if err != nil {
+			handleErrorFunc(err)
+			return
+		}
+
+		if len(crds) == 0 {
+			handleErrorFunc(apierrors.NewNotFound(schema.GroupResource{Group: requestInfo.APIGroup, Resource: "customresourcedefinitions"}, ""))
```

Comment:
shouldn't you fall through to the delegate?

Reply:
do we actually need this if? The loop below will just fall through and do the same.